BoxContentSDK/BoxContentSDK/Models/BOXItem.m

```diff
@@ -248,7 +248,9 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
                                                                         nullAllowed:YES];
         NSMutableArray *collections = [NSMutableArray arrayWithCapacity:collectionsJSONArray.count];
         for (NSDictionary *dict in collectionsJSONArray) {
-            [collections addObject:[[BOXCollection alloc] initWithJSON:dict]];
+            if (![dict isEqual:[NSNull null]]) {
+                [collections addObject:[[BOXCollection alloc] initWithJSON:dict]];
```

Comment:
Do you think, while we're at it, we might want to check that we're not getting `nil` from `initWithJSON:`? Because adding that to an array would also crash.

Reply:
As far as I can tell, that can't happen unless we change our code to make it happen, which would be caught by unit tests. Is that understanding inaccurate?